core/src/main/java/io/grpc/StreamTracer.java

```diff
@@ -62,32 +62,45 @@ public void inboundMessage() {
   }
 
   /**
+   * An outbound message has been completely serialized, revealing its wire size.
+   */
+  public void outboundMessageSerialized(long wireSize) {
+  }
+
+  /**
+   * An inbound message has been completely read from the wire, revealing its wire size.
+   */
+  public void inboundMessageReceived(long wireSize) {
```

Comment:
This is even more problematic with stream-wide compression, since we can't easily correlate compressed bytes to decompressed bytes (except if we decompressed one byte at a time, which is not acceptable).